test/framework/src/main/java/org/opensearch/test/OpenSearchTestCase.java

```diff
@@ -626,6 +627,7 @@ public void log(StatusData data) {
     protected static void checkStaticState(boolean afterClass) throws Exception {
         if (afterClass) {
             MockPageCacheRecycler.ensureAllPagesAreReleased();
+            StrictCheckSpanProcessor.validateTracingStateOnShutdown();
```

Comment:
This should go to `OpenSearchIntegTestCase` - this test case adds mock telemetry & co, `OpenSearchTestCase` known nothing about that

Reply:
I just clicked on the checkStaticState method and didnt realize that its taking me to different class. :)